src/Dapr.AspNetCore/CloudEventsMiddleware.cs

```diff
@@ -83,17 +96,43 @@ private async Task ProcessBodyAsync(HttpContext httpContext, string charSet)
             string contentType;
 
             // Check whether to use data or data_base64 as per https://github.com/cloudevents/spec/blob/v1.0.1/json-format.md#31-handling-of-data
-            var isDataSet = json.TryGetProperty("data", out var data);
-            var isBinaryDataSet = json.TryGetProperty("data_base64", out var binaryData);
+            // Get the property names by OrdinalIgnoreCase comparison to support case insensitive JSON as the Json Serializer for AspCore already supports it by default.
+            var jsonPropNames = json.EnumerateObject().ToArray();
+
+            var dataPropName = jsonPropNames
+                .Select(d => d.Name)
+                .FirstOrDefault(d => d.Equals(CloudEventPropertyNames.Data, StringComparison.OrdinalIgnoreCase));
+
+            var dataBase64PropName = jsonPropNames
+                .Select(d => d.Name)
+                .FirstOrDefault(d =>
+                    d.Equals(CloudEventPropertyNames.DataBase64, StringComparison.OrdinalIgnoreCase));
+
+            var isDataSet = false;
+            var isBinaryDataSet = false;
+            JsonElement data = default;
+
+            if (!string.IsNullOrWhiteSpace(dataPropName))
```

Comment:
Nit: as the name of the property is hidden behind a constant, we don't known, here, for certain, that whitespace doesn't indicate a valid property.  Testing only for `null` or non-`null` seems more appropriate.

Reply:
I can't say I agree on this one, first as you said its a known value stored in a constant, which is immutable, it cant be changed other than someone in a PR, which should be alarming to be honest.
And generally when dealing with strings, its a best practice to check against all the three possible "empty" values, null, empty string and whitespace character, which is what `IsNullOrWhiteSpace` does.
That said, it wont affect the functionality in this case so I wont die over this, but it will need to be updated when the project starts using nullable ref types, which is something that surprised me. I would expect it to already be using them since its targeting .NET 6+.

Reply:
> And generally when dealing with strings, its a best practice to check against all the three possible "empty" values, null, empty string and whitespace character, which is what IsNullOrWhiteSpace does.

In this case, however, both the empty string and a white-space-only string are technically valid property names that could be returned by the prior search.  We know ourselves that those values are unlikely to be the actual names of the property being searched for but, as they're hidden behind the constant, the test should not make such assumptions.

> I would expect it to already be using them since its targeting .NET 6+.

Adding nullable annotations has been in the issues list for a while, but it's a pretty sizable engineering task.  The switch to .NET 6/7 is fairly recent so it will take time to start using the more modern language conveniences a large scale.

Reply:
> > And generally when dealing with strings, its a best practice to check against all the three possible "empty" values, null, empty string and whitespace character, which is what IsNullOrWhiteSpace does.
> 
> In this case, however, both the empty string and a white-space-only string are technically valid property names that could be returned by the prior search. We know ourselves that those values are unlikely to be the actual names of the property being searched for but, as they're hidden behind the constant, the test should not make such assumptions.
> 
I agree on the principle, however the value being in the constant compared to being in a string visible right there makes no difference to me. In both cases it is hardcoded in the codebase and immutable in runtime. If a value has matched and returns it will be the same as the value in the constant, with the only difference on its casing, which is ignored. It cannot return an empty string or whitespace. The only difference is this way it can be reused in other places.

> > I would expect it to already be using them since its targeting .NET 6+.
> 
> Adding nullable annotations has been in the issues list for a while, but it's a pretty sizable engineering task. The switch to .NET 6/7 is fairly recent so it will take time to start using the more modern language conveniences a large scale.

Yeah that is a valid point, it does require quite a bit of refactoring to eliminate all the warnings. And there's quite a few possible NRE already in there.

Reply:
Alright as I see that this is not treated as a Nit at all, I'll go ahead  and refactor it. However I will leave it here that the argument holds no ground as the purpose of constants is to reuse static values that do not mutate during runtime. And that the whole purpose of those two blocks of code is to grab either the data or data64 properties of the CloudEvent. If anything else returns for whatever reason, the middleware is critically broken.